src/index.ts

```diff
@@ -1,49 +1,23 @@
 import * as core from "@actions/core";
-import { readConfig } from "./config";
-import { apiClient } from "./client/apiClient";
-import { isSuccessful } from "./run/status";
-import { startRun } from "./run/start";
-import { FinishedRun, waitForRunEnd } from "./run/ongoing";
 
-async function run(): Promise<void> {
-  try {
-    const config = readConfig();
-    const client = apiClient(config.api);
-
-    await client.checkCloudCompatibility();
-
-    const startedRun = await startRun(client, config);
-    core.setOutput("run_id", startedRun.runId);
-    core.setOutput("reports_url", startedRun.reportsUrl);
-    core.setOutput("runs_url", startedRun.runsUrl);
-
-    const finishedRun = await waitForRunEnd(client, startedRun);
-    logResult(finishedRun);
-
-    // TODO RND-7 Metrics summary and XML report
-
-    core.setOutput("runs_status_code", finishedRun.statusCode);
-    core.setOutput("runs_status_name", finishedRun.statusName);
-    core.setOutput("run_assertions", finishedRun.assertions);
-  } catch (error) {
-    const msg = error instanceof Error ? error.message : `Unknown error: ${error}`;
-    core.setFailed(msg);
-  }
-}
-
-const logResult = (finishedRun: FinishedRun) => {
-  if (isSuccessful(finishedRun.statusCode)) {
-    core.info(`Run ${finishedRun.runId} finished with status ${finishedRun.statusName}`);
-  } else {
-    core.setFailed(`Run ${finishedRun.runId} failed with status ${finishedRun.statusName}`);
-  }
-
-  for (const assertion of finishedRun.assertions) {
-    if (assertion.result) {
-      core.info(`${assertion.message} succeeded with value ${assertion.actualValue}`);
-    } else {
-      core.error(`${assertion.message} failed with value ${assertion.actualValue}`);
-    }
+import runMain from "./runMain";
+import runCleanup from "./runCleanup";
```

Comment:
perhaps producing 2 different js file to be able to configure the action with 2 different endpoints?

Reply:
The action used for publishing only handles a single JS bundle file and we didn't want to invest time to customize it.